Investigate permanent UI test failure in SmokeTest#customTrackingProtectionSettingsTest and SmokeTest#testStrictVisitSheetDetails

**customTrackingProtectionSettingsTest**

```Kotlin
androidx.test.espresso.PerformException: Error performing 'single click' on view '(view.getId() is <2131297086/org.mozilla.fenix.debug:id/menuButton>)'.
at androidx.test.espresso.PerformException$Builder.build(PerformException.java:1)
at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:9)
at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:7)
at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:8)
at androidx.test.espresso.ViewInteraction.desugaredPerform(ViewInteraction.java:11)
at androidx.test.espresso.ViewInteraction.perform(ViewInteraction.java:4)
at org.mozilla.fenix.ui.robots.HomeScreenRobot$Transition.openThreeDotMenu(HomeScreenRobot.kt:370)
at org.mozilla.fenix.ui.SmokeTest.customTrackingProtectionSettingsTest(SmokeTest.kt:412)
... 58 trimmed
Caused by: java.lang.RuntimeException: Action will not be performed because the target view does not match one or more of the following constraints:
(view has effective visibility <VISIBLE> and view.getGlobalVisibleRect() covers at least <90> percent of the view's area)
Target view: "AppCompatImageView{id=2131297086, res-name=menuButton, visibility=VISIBLE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@e99749e, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0}"
at androidx.test.espresso.ViewInteraction.doPerform(ViewInteraction.java:20)
at androidx.test.espresso.ViewInteraction.access$100(ViewInteraction.java:1)
at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:2)
at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:1)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at android.os.Handler.handleCallback(Handler.java:938)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:223)
at android.app.ActivityThread.main(ActivityThread.java:7656)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

**testStrictVisitSheetDetails**


```Kotlin
androidx.test.espresso.base.DefaultFailureHandler$AssertionFailedWithCauseError: '(view has effective visibility <VISIBLE> and view.getGlobalVisibleRect() to return non-empty rectangle)' doesn't match the selected view.
Expected: (view has effective visibility <VISIBLE> and view.getGlobalVisibleRect() to return non-empty rectangle)
Got: view.getVisibility() was <GONE>
View Details: AppCompatTextView{id=2131296729, res-name=cryptominers, visibility=GONE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=true, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@a086a1, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, text=Cryptominers, input-type=0, ime-target=false, has-links=false}
at dalvik.system.VMStack.getThreadStackTrace(Native Method)
at java.lang.Thread.getStackTrace(Thread.java:1736)
at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:12)
at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:7)
at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:5)
at androidx.test.espresso.ViewInteraction.check(ViewInteraction.java:12)
at org.mozilla.fenix.ui.robots.EnhancedTrackingProtectionRobotKt.assertCryptominersBlocked(EnhancedTrackingProtectionRobot.kt:177)
at org.mozilla.fenix.ui.robots.EnhancedTrackingProtectionRobotKt.access$assertCryptominersBlocked(EnhancedTrackingProtectionRobot.kt:1)
at org.mozilla.fenix.ui.robots.EnhancedTrackingProtectionRobot.verifyCryptominersBlocked(EnhancedTrackingProtectionRobot.kt:47)
at org.mozilla.fenix.ui.StrictEnhancedTrackingProtectionTest$testStrictVisitSheetDetails$8.invoke(StrictEnhancedTrackingProtectionTest.kt:186)
at org.mozilla.fenix.ui.StrictEnhancedTrackingProtectionTest$testStrictVisitSheetDetails$8.invoke(StrictEnhancedTrackingProtectionTest.kt:183)
at org.mozilla.fenix.ui.robots.EnhancedTrackingProtectionRobot$Transition.openDetails(EnhancedTrackingProtectionRobot.kt:107)
at org.mozilla.fenix.ui.StrictEnhancedTrackingProtectionTest.testStrictVisitSheetDetails(StrictEnhancedTrackingProtectionTest.kt:183)
... 33 trimmed
```

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21748)

💥 `customTrackingProtectionSettingsTest` failed 1x on 10/6 Main 
[Firebase link](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/6970110157146624558/executions/bs.f64400300f08c872/testcases/3/test-cases)
[Firebase link](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/6970110157146624558/executions/bs.87ed7a075edc75ba/testcases/3/test-cases)

androidx.test.espresso.PerformException: Error performing 'single click' on view '(view.getId() is <2131297086/org.mozilla.fenix.debug:id/menuButton>)'.
	at androidx.test.espresso.PerformException$Builder.build(PerformException.java:1)
	at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:9)
	at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:7)
	at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:8)
	at androidx.test.espresso.ViewInteraction.desugaredPerform(ViewInteraction.java:11)
	at androidx.test.espresso.ViewInteraction.perform(ViewInteraction.java:4)
	at org.mozilla.fenix.ui.robots.HomeScreenRobot$Transition.openThreeDotMenu(HomeScreenRobot.kt:370)
	at org.mozilla.fenix.ui.SmokeTest.customTrackingProtectionSettingsTest(SmokeTest.kt:412)
	... 58 trimmed
Caused by: java.lang.RuntimeException: Action will not be performed because the target view does not match one or more of the following constraints:
(view has effective visibility <VISIBLE> and view.getGlobalVisibleRect() covers at least <90> percent of the view's area)
Target view: "AppCompatImageView{id=2131297086, res-name=menuButton, visibility=VISIBLE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@8228023, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0}"
	at androidx.test.espresso.ViewInteraction.doPerform(ViewInteraction.java:20)
	at androidx.test.espresso.ViewInteraction.access$100(ViewInteraction.java:1)
	at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:2)
	at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:1)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7656)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
    

💥 `testStrictVisitSheetDetails ` failed 1x on 10/6 Main 
[Firebase link](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/6970110157146624558/executions/bs.efd7653fba6e4286/testcases/3/test-cases)
[Firebase link](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/6970110157146624558/executions/bs.90489d6d77bd04d5/testcases/3/test-cases)

androidx.test.espresso.base.DefaultFailureHandler$AssertionFailedWithCauseError: '(view has effective visibility <VISIBLE> and view.getGlobalVisibleRect() to return non-empty rectangle)' doesn't match the selected view.
Expected: (view has effective visibility <VISIBLE> and view.getGlobalVisibleRect() to return non-empty rectangle)
     Got: view.getVisibility() was <GONE>
View Details: AppCompatTextView{id=2131296729, res-name=cryptominers, visibility=GONE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=true, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@1265e90, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, text=Cryptominers, input-type=0, ime-target=false, has-links=false}

	at dalvik.system.VMStack.getThreadStackTrace(Native Method)
	at java.lang.Thread.getStackTrace(Thread.java:1736)
	at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:12)
	at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:7)
	at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:5)
	at androidx.test.espresso.ViewInteraction.check(ViewInteraction.java:12)
	at org.mozilla.fenix.ui.robots.EnhancedTrackingProtectionRobotKt.assertCryptominersBlocked(EnhancedTrackingProtectionRobot.kt:177)
	at org.mozilla.fenix.ui.robots.EnhancedTrackingProtectionRobotKt.access$assertCryptominersBlocked(EnhancedTrackingProtectionRobot.kt:1)
	at org.mozilla.fenix.ui.robots.EnhancedTrackingProtectionRobot.verifyCryptominersBlocked(EnhancedTrackingProtectionRobot.kt:47)
	at org.mozilla.fenix.ui.StrictEnhancedTrackingProtectionTest$testStrictVisitSheetDetails$8.invoke(StrictEnhancedTrackingProtectionTest.kt:186)
	at org.mozilla.fenix.ui.StrictEnhancedTrackingProtectionTest$testStrictVisitSheetDetails$8.invoke(StrictEnhancedTrackingProtectionTest.kt:183)
	at org.mozilla.fenix.ui.robots.EnhancedTrackingProtectionRobot$Transition.openDetails(EnhancedTrackingProtectionRobot.kt:107)
	at org.mozilla.fenix.ui.StrictEnhancedTrackingProtectionTest.testStrictVisitSheetDetails(StrictEnhancedTrackingProtectionTest.kt:183)
	... 33 trimmed
Caused by: junit.framework.AssertionFailedError: '(view has effective visibility <VISIBLE> and view.getGlobalVisibleRect() to return non-empty rectangle)' doesn't match the selected view.
Expected: (view has effective visibility <VISIBLE> and view.getGlobalVisibleRect() to return non-empty rectangle)
     Got: view.getVisibility() was <GONE>
View Details: AppCompatTextView{id=2131296729, res-name=cryptominers, visibility=GONE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=true, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@1265e90, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, text=Cryptominers, input-type=0, ime-target=false, has-links=false}

	at androidx.test.espresso.matcher.ViewMatchers.assertThat(ViewMatchers.java:16)
	at androidx.test.espresso.assertion.ViewAssertions$MatchesViewAssertion.check(ViewAssertions.java:11)
	at androidx.test.espresso.ViewInteraction$SingleExecutionViewAssertion.check(ViewInteraction.java:2)
	at androidx.test.espresso.ViewInteraction$2.call(ViewInteraction.java:12)
	at androidx.test.espresso.ViewInteraction$2.call(ViewInteraction.java:1)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7656)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)

--- a/app/src/androidTest/assets/pages/trackingPage.html
+++ b/app/src/androidTest/assets/pages/trackingPage.html
@@ -13,31 +13,31 @@
     <h3>Level 1 (Basic) List</h3>
     <p>social-track-digest256:</p>
     <img
-            src="https://social-track-digest256.dummytracker.org/test_not_blocked.png"
-            onerror="this.onerror=null;this.src='https://not-a-tracker.dummytracker.org/test_blocked.png'">
+            src="https://social-track-digest256.dummytracker.org/test_not_blocked.png" alt="social not blocked"
+            onerror="this.onerror=null;this.src='https://not-a-tracker.dummytracker.org/test_blocked.png';this.alt='social blocked'">
     <br/>
     <p>ads-track-digest256:</p>
     <img
             src="https://ads-track-digest256.dummytracker.org/test_not_blocked.png"
-            onerror="this.onerror=null;this.src='https://not-a-tracker.dummytracker.org/test_blocked.png'">
+            onerror="this.onerror=null;this.src='https://not-a-tracker.dummytracker.org/test_blocked.png';this.alt='ads blocked'">
     <br/>
     <p>analytics-track-digest256:</p>
     <img
             src="https://analytics-track-digest256.dummytracker.org/test_not_blocked.png"
-            onerror="this.onerror=null;this.src='https://not-a-tracker.dummytracker.org/test_blocked.png'">
+            onerror="this.onerror=null;this.src='https://not-a-tracker.dummytracker.org/test_blocked.png';this.alt='analytics blocked'">
     <br/>
     <p>Fingerprinting:
     <pre id="result">test not run</pre>
     <script src="https://base-fingerprinting-track-digest256.dummytracker.org/tracker.js"
-            onerror="this.onerror=null;var result=document.getElementById('result');result.innerHTML='blocked';"
-            onload="this.onload=null;var result=document.getElementById('result');result.innerHTML='NOT blocked';"
+            onerror="this.onerror=null;var result=document.getElementById('result');result.innerHTML='Fingerprinting blocked';"
+            onload="this.onload=null;var result=document.getElementById('result');result.innerHTML='Fingerprinting NOT blocked';"
     ></script>
     </p>
     <br/>
     <p>Cryptomining:
       <img
-              src="https://base-cryptomining-track-digest256.dummytracker.org/test_not_blocked.png" alt="not blocked"
-              onerror="this.onerror=null;this.src='https://not-a-tracker.dummytracker.org/test_blocked.png';this.alt='blocked'">
+              src="https://base-cryptomining-track-digest256.dummytracker.org/test_not_blocked.png" alt="Cryptomining not blocked"
+              onerror="this.onerror=null;this.src='https://not-a-tracker.dummytracker.org/test_blocked.png';this.alt='Cryptomining blocked'">
     </p>
 
     <p><b>Cookie blocking</b>

--- a/app/src/androidTest/java/org/mozilla/fenix/ui/EnhancedTrackingProtectionTest.kt
+++ b/app/src/androidTest/java/org/mozilla/fenix/ui/EnhancedTrackingProtectionTest.kt
@@ -8,7 +8,6 @@ import androidx.core.net.toUri
 import okhttp3.mockwebserver.MockWebServer
 import org.junit.After
 import org.junit.Before
-import org.junit.Ignore
 import org.junit.Rule
 import org.junit.Test
 import org.mozilla.fenix.customannotations.SmokeTest
@@ -165,7 +164,6 @@ class EnhancedTrackingProtectionTest {
         }
     }
 
-    @Ignore("Permanent failure: https://github.com/mozilla-mobile/fenix/issues/27312")
     @Test
     fun testStrictVisitSheetDetails() {
         appContext.settings().setStrictETP()
@@ -182,7 +180,11 @@ class EnhancedTrackingProtectionTest {
 
         navigationToolbar {
         }.enterURLAndEnterToBrowser(trackingProtectionTest.url) {
-            verifyTrackingProtectionWebContent("blocked")
+            verifyTrackingProtectionWebContent("social blocked")
+            verifyTrackingProtectionWebContent("ads blocked")
+            verifyTrackingProtectionWebContent("analytics blocked")
+            verifyTrackingProtectionWebContent("Fingerprinting blocked")
+            verifyTrackingProtectionWebContent("Cryptomining blocked")
         }
         enhancedTrackingProtection {
         }.openEnhancedTrackingProtectionSheet {
@@ -197,7 +199,6 @@ class EnhancedTrackingProtectionTest {
         }
     }
 
-    @Ignore("Permanent failure: https://github.com/mozilla-mobile/fenix/issues/27312")
     @SmokeTest
     @Test
     fun customTrackingProtectionSettingsTest() {
@@ -217,7 +218,13 @@ class EnhancedTrackingProtectionTest {
             // browsing a basic page to allow GV to load on a fresh run
         }.enterURLAndEnterToBrowser(genericWebPage.url) {
         }.openNavigationToolbar {
-        }.enterURLAndEnterToBrowser(trackingPage.url) {}
+        }.enterURLAndEnterToBrowser(trackingPage.url) {
+            verifyTrackingProtectionWebContent("social blocked")
+            verifyTrackingProtectionWebContent("ads blocked")
+            verifyTrackingProtectionWebContent("analytics blocked")
+            verifyTrackingProtectionWebContent("Fingerprinting blocked")
+            verifyTrackingProtectionWebContent("Cryptomining blocked")
+        }
 
         enhancedTrackingProtection {
         }.openEnhancedTrackingProtectionSheet {

